docs/pages/07_service.adoc

```diff
@@ -0,0 +1,1092 @@
+----
+
+In the example above, the `getEmployee` function may return a SalesPersonTransfer or a DeveloperTransfer transfer object. Within the body of the function, it can be decided whether a SalesPersonTransfer or DeveloperTransfer object is returned.
+
+
+////
+The optional `static` keyword is used to create service that can be invoked without an existing instance of the transfer object. Static methods cannot use any of the fields or derived fields of the transfer object in which they were defined. In other words, static actions use the transfer objects that contain them as an embedding namespace.
+////
+
+
+== AuthN and AuthZ
+
+AuthN (Authentication) is the process of verifying a user’s identity and AuthZ (Authorization) is the process of determining whether an authenticated user has access to a particular resource or service. Authorization is critical for ensuring that only authorized users have access to sensitive data and resources. The following chapters describe JUDO's authentication and authorization process.
+
+=== Actor
+
+An actor represents a generic user of the system and can be used to represent humans or another system.
+
+To define an actor, use the `actor` keyword.
+
+*Syntax:*
+
+----
+actor <name> [(<entity> <mapping-field>)]
+    [realm "<realm>"]
+    [identity <identity>];
+----
+
+where the <name> is the name of the actor. The <entity> is the entity type of the linked entity instance. The <mapping-field> is the name of the field that refers to the linked entity instance. The linked entity instance is the persistent representation of the user in the application. If there is no linked entity, the user is not represented in the application. A typical case of not represented user is an anonymous user, who shall not be identified within the application.
+
+The <realm> is the name of the security policy domain attached to the actor. The realm is collection of users who can be authenticated to use the application. The realms and identity management used by JUDO are out of scope of the application. The default realm management application that can be used by a JUDO application is Keycloack, but any OpenID3 compatible identity management application can be easily integrated.
+
+The <identity> is an expression that selects an identifier field of the mapped <entity>. The <identity> is used to unambiguously identify the user within the application (e.g. find only one instance of the mapped entity) to define the relationship with the user in the <realm>.
+
+// TODO: managed user missing
+
+[NOTE]
+====
+There is an alternative syntax to define an actor. This alternative is a bit more expressive, although it may be too verbose.
+
+----
+actor <name> maps <entity> as <mapping-field>
+    [realm "<realm>"]
+    [identity <identity>];
+----
+
+====
+
+
+*Example:*
+
+----
+actor SalesPersonActor(SalesPerson salesPerson)
+    realm "COMPANY"
+    identity salesPerson.email ;
+----
+
```

Comment:
Are we sure that we don't need the `claims` concept?

Reply:
@claim shall select which attribute of the access token is used for identification

Reply:
@Claim("USERNAME")
actor SalesPersonActor(SalesPerson salesPerson) realm "COMPANY" identity salesPerson.email ;